app/controllers/sessions_controller.rb

```diff
@@ -19,6 +16,13 @@ def create
 
   private
 
+  def find_user(session)
+    who, pass = session.fetch(:who), session.fetch(:password)
+    if who.is_a?(String) && pass.is_a?(String)
```

Comment:
👍 Just noting here that `session.permit(:who)` would ensure `who` is in a list of [permitted scalars](https://github.com/rails/rails/blob/4-2-stable/actionpack/lib/action_controller/metal/strong_parameters.rb#L510-L531), such as String, but there's no way to restrict the list to just `String`, so we go type-checking route ourselves.